dashboard/dash2.js

```diff
@@ -0,0 +1,134 @@
+// dashboard.js must have been loaded before this one.
+
+"use strict";
+
+// Colors for annotations
+let col_DOWN = "tomato"             // a bit better contrast than "red"
+let col_WORKINGHARD = "deepskyblue" // 75%
+let col_WORKING = "lightskyblue"    // 50%
+let col_COASTING = "lightcyan"      // 25%
+
+// Where to go when clicking on a row, this is opened with ?host=<hostname> from the JSON datum
+let machine_page = `machine-detail.html`
+
+let timeout_minutes = 5;
+
+// At-a-glance data are generated by naicreport, every 5min ideally.
+let fields = [
+    // fqdn or prefix; fqdn is preferred
+    {name: "Host", tag: "hostname"},
+
+    // From the last record on the host.  0=ok, 1=error (maybe more codes later).
+    {name: "CPU\nstatus", tag: "cpu_status"},
+    {name: "GPU\nstatus", tag: "gpu_status"},
+
+    // Unique users in the period.  This will never be greater than jobs; a user can have
+    // several jobs, but not zero, and jobs can only have one user.
+    {name: "Users\n(recent)", tag: "users_recent"},
+    {name: "Users\n(12h)", tag: "users_12h"},
+
+    // Unique jobs running within the period.  The "recent" values are for the last 30min.
+    {name: "Jobs\n(recent)", tag: "jobs_recent"},
+    {name: "Jobs\n(12h)", tag: "jobs_12h"},
+
+    // Relative to system information.  The 12h numbers are averages in a moving
+    // window, the "recent" numbers are averages in a 30m window.
+    {name: "CPU%\n(recent)", tag: "cpu_recent"},
+    {name: "CPU%\n(12h)", tag: "cpu_12h"},
+    {name: "Mem%\n(recent)", tag: "mem_recent"},
+    {name: "Mem%\n(12h)", tag: "mem_12h"},
+    {name: "GPU%\n(recent)", tag: "gpu_recent"},
+    {name: "GPU%\n(12h)", tag: "gpu_12h"},
+    {name: "GPUMEM%\n(recent)", tag: "gpumem_recent"},
+    {name: "GPUMEM%\n(12h)", tag: "gpumem_12h"},
+
+    // Number of new hogs and zombies encountered in the period, as of the last
+    // generated report.  This currently changes rarely, as the reports are run
+    // every 12h.
+    {name: "Hogs\n(24h)", tag: "hogs_24h"},
+    {name: "Zombies\n(24h)", tag: "zombies_24h"},
+]
+
+// Compute field offsets in field table
+let offs = {};
+(function () {
+    let i = 0
+    for (let f of fields ) {
+        offs[f.tag] = i
+	i++
+    }
+})();
+
+let reload_timer = null;
```

Comment:
And here and directly above, but the one above the init function is necessary.  Actually the init function can just be a block now that we have let?

Reply:
Use for .. in to step by index, then we have i, and use that to avoid the problem altogether.